### DISCUSSION TITLE
Why did a branch in a fork show up in the original repo?

### DISCUSSION THREAD

**Discussion by egil** [community/community#22789] [New to GitHub]
<p>Hi all,</p>
<p>Somebody created a fork of my repo (egil/bUnit) and then created a branch in their fork. Now I see the branch show up in my repo, and I wonder how that happened. The branch in question is <a href="https://github.com/egil/bUnit/tree/Packt-edits" rel="nofollow noopener">https://github.com/egil/bUnit/tree/Packt-edits</a>.</p>
<p>Did I do something myself? Or do I have some security settings that are too loose?</p>
<p>Thanks, Egil</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hi <a class="mention" href="/egil">@egil</a> :wave:</p>
<p>I know it’s been a while since you originally posted and it looks like that branch has since been deleted. I did want to shed some light for you on the subject nonetheless.</p>
<p>On GitHub, commits made in forks are stored in a network that is shared with the parent repository. This is so that any commit that exists in both the parent repository and a fork only has to be stored in our database once. However, it also means that a commit that exists only in a fork can still be viewed via a URL containing the username that corresponds to the parent repository.</p>
<p>That branch would have been created when a commit from that user had been merged into your repository.</p>

**Comment by egil** [comment]
<p>Thanks for getting back to me Ernest.</p>
<p>I think that makes sense, but the reason I asked is because I am pretty close to 100% sure that no commit was merged in to my repo, neither by me or by the other user, who does not have access rights to do so.</p>
<p>Anyway, I have since deleted the extra branch as our collaboration for now is done.</p>
